athenet/sparsifying/utils/interval.py

```diff
@@ -0,0 +1,284 @@
+"""Intervals in Theano including special functions for sparsifying."""
+
+from athenet.sparsifying.utils.numlike import Numlike
+from theano import function
+from theano import tensor as T
+from theano.ifelse import ifelse
+
+
+class Interval(Numlike):
+    """Theano interval class
+
+    Note: Should be treated as interval type with bounds as Theano nodes.
+    Opetations on Interval create nodes in Theano graph."""
+
+    def __init__(self, lower, upper):
+        """Creates interval.
+
+        :lower: lower bound of Interval to be set
+        :upper: upper bound if Interval to be set
+
+        Note: lower must be lower than upper. It is not being checked."""
+        self.lower = lower
+        self.upper = upper
+
+    def __getitem__(self, at):
+        """Returns specified slice of interval as a interval.
```

Comment:
Maybe write somewhere that Interval might be multidimensional and what do this mean (like "stores many Intervals in one object")